### DISCUSSION TITLE
To find URLs on content pages  of an entire site collection

### DISCUSSION THREAD

**Discussion by AnjalyDeepak** [pnp/powershell#290] [Q&A]
Is there any Powershell Script to find specific URLs present anywhere on content pages of an entire site collection site pages?

After migrating the pages and contents to the modern site, in some cases, I still have some document and page links to our old intranet. Basically, these links are on pages. I need to check all the pages and identify if there's anything that points to the old intranet. Is there a way to somehow automatically identify these links on all my subsites using PnP Powershell?

**Comment by erwinvanhunen** [comment]
As a modern page is basically a set of webparts, you'll have to code that yourself. Think of using cmdlets like `Get-PnPPageComponent`, then you would have to loop through the results and look at the `PropertiesJson` property of the components. As each webpart stores content differently in the json maybe the easiest would be to just to a text search. Something alike this:

```powershell
Get-PnPPageComponent  -Page $pagetitle | %{$_.PropertiesJson | Select-String https://linktoyouroldintranet}
```

If that returns something you'll know there is a link on that page pointing to the old intranet.

Obviously you'll have to loop through all the pages, so do something alike

```powershell
$pages = Get-PnPListItem -List SitePages | %{$_["Title"]}
```

This will select all the page titles. Then using the standard PowerShell constructs you can do things like looping through that collection and retrieving the correct page, etc. etc.

**Reply by AnjalyDeepak** [reply]
Thank you for the solution. But I get some error while executing the command _Get-PnPPageComponent_ 
```
Get-PnPPageComponent : The term 'Get-PnPPageComponent' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if 
a path was included, verify that the path is correct and try again.
```
The versions of _SharePointPnPPowerShell_ and _Microsoft.Online.SharePoint.PowerShell_ installed are:
```
Name                                   Version         
----                                   -------         
SharePointPnPPowerShellOnline          3.24.2008.0     
Microsoft.Online.SharePoint.PowerShell 16.0.20324.12000
Microsoft.Online.SharePoint.PowerShell 16.0.20324.0    
```

**Reply by ToddKlindt** [reply]
The command Get-PnPPageComponent is in the PnP.PowerShell module. Do you have that installed? In the SharePointPnPPowerShellOnline  module the command is Get-PnPClientSideComponent. It may or may not work like @erwinvanhunen shows above with the other command. If possible, you should transition to PnP.PowerShell.
